nni/experiment/config/v1.py

```diff
@@ -0,0 +1,201 @@
+    _move_field(v1, v2, 'logLevel', 'log_level')
+    _drop_field(v1, 'logCollection')
+    _drop_field(v1, 'useAnnotation')  # FIXME
+
+    if 'trial' in v1:
+        v1_trial = v1.pop('trial')
+        _move_field(v1_trial, v2, 'command', 'trial_command')
+        _move_field(v1_trial, v2, 'codeDir', 'trial_code_directory')
+        _move_field(v1_trial, v2, 'gpuNum', 'trial_gpu_number')
+
+    for algo_type in ['tuner', 'assessor', 'advisor']:
+        if algo_type not in v1:
+            continue
+        v1_algo = v1.pop(algo_type)
+
+        builtin_name = v1_algo.pop(f'builtin{algo_type.title()}Name', None)
+        class_args = v1_algo.pop('classArgs', None)
+
+        if builtin_name is not None:
+            v2_algo = AlgorithmConfig(name=builtin_name, class_args=class_args)
+            # FIXME: check built-in name available
+
+        else:
+            class_directory = util.canonical_path(v1_algo.pop('codeDir'))
+            class_file_name = v1_algo.pop('classFileName')
+            assert class_file_name.endswith('.py')
+            class_name = class_file_name[:-3] + '.' + v1_algo.pop('className')
+            v2_algo = CustomAlgorithmConfig(
+                class_name=class_name,
+                class_directory=class_directory,
+                class_args=class_args
+            )
+
+        setattr(v2, algo_type, v2_algo)
+        _drop_field(v1_algo, 'includeIntermediateResults')  # FIXME
+        _move_field(v1_algo, v2, 'gpuIndices', 'tuner_gpu_indices')
+        assert not v1_algo, v1_algo
+
+    ts = v2.training_service
+
+    if platform == 'local':
+        local_config = v1.pop('localConfig', {})
+        _move_field(local_config, ts, 'gpuIndices', 'gpu_indices')
+        _move_field(local_config, ts, 'maxTrialNumPerGpu', 'max_trial_number_per_gpu')
+        _move_field(local_config, ts, 'useActiveGpu', 'use_active_gpu')
+        assert not local_config, local_config
+
+    if platform == 'remote':
+        remote_config = v1.pop('remoteConfig', {})
+        _move_field(remote_config, ts, 'reuse', 'reuse_mode')
+        assert not remote_config, remote_config
+
+        ts.machine_list = []
+        for v1_machine in v1.pop('machineList'):
+            v2_machine = RemoteMachineConfig()
+            _move_field(v1_machine, v2_machine, 'ip', 'host')
+            _move_field(v1_machine, v2_machine, 'port', 'port')
+            _move_field(v1_machine, v2_machine, 'username', 'user')
+            v2_machine.ssh_key_file = util.canonical_path(v1_machine.pop('sshKeyPath', None))
+            _move_field(v1_machine, v2_machine, 'passphrase', 'ssh_passphrase')
+            _move_field(v1_machine, v2_machine, 'gpuIndices', 'gpu_indices')
+            _move_field(v1_machine, v2_machine, 'maxTrialNumPerGpu', 'max_trial_number_per_gpu')
+            _move_field(v1_machine, v2_machine, 'useActiveGpu', 'use_active_gpu')
+            _move_field(v1_machine, v2_machine, 'preCommand', 'trial_prepare_command')
```

Comment:
we just rename this field to `pythonPath` in https://github.com/microsoft/nni/pull/3367#issuecomment-783125093